packages/nimble-components/src/table-column/mapping/index.ts

```diff
@@ -0,0 +1,334 @@
+/* eslint-disable max-classes-per-file */
+import { DesignSystem } from '@microsoft/fast-foundation';
+import {
+    attr,
+    booleanConverter,
+    Notifier,
+    nullableNumberConverter,
+    Observable,
+    observable,
+    Subscriber
+} from '@microsoft/fast-element';
+import { styles } from './styles';
+import { template } from './template';
+import type { TableAnyField } from '../../table/types';
+import { TableColumn } from '../base';
+import { TableColumnSortOperation, TableColumnValidity } from '../base/types';
+import { tableColumnMappingCellViewTag } from './cell-view';
+import { Mapping } from './mappings';
+import { tableColumnMappingGroupHeaderViewTag } from './group-header-view';
+import { mixinGroupableColumnAPI } from '../mixins/groupable-column';
+import { mixinFractionalWidthColumnAPI } from '../mixins/fractional-width-column';
+import { mixinFixedWidthColumnAPI } from '../mixins/fixed-width-column';
+import { MappingText } from './mappings/text';
+import { MappingSpinner } from './mappings/spinner';
+import { MappingIcon } from './mappings/icon';
+import type { ColumnInternalsOptions } from '../base/models/column-internals';
+import {
+    iconColumnValidityFlagNames,
+    mappingColumnValidityFlagNames,
+    TableColumnIconValidator,
+    TableColumnMappingValidator
+} from './models/column-validator';
+
+export type TableColumnMappingCellRecord = TableAnyField<'value'>;
+export interface TableColumnMappingColumnConfig {
+    typedKeysToMappings: (
+        | readonly [number | null, Mapping]
+        | readonly [boolean | null, Mapping]
+        | readonly [string | null, Mapping]
+    )[];
+}
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-table-column-mapping': TableColumnMapping;
+        'nimble-table-column-icon': TableColumnIcon;
+    }
+}
+
+/**
+ * Base class for table columns that map values to content
+ */
+export abstract class TableColumnMappingBase
+    extends TableColumn<TableColumnMappingColumnConfig>
+    implements Subscriber {
+    @attr({ attribute: 'field-name' })
+    public fieldName?: string;
+
+    @attr({ attribute: 'key-type' })
+    public keyType: 'string' | 'number' | 'boolean' = 'string';
+
+    /** @internal */
+    @observable
+    public mappings?: Mapping[];
+
+    protected abstract supportedMappingTypes: (typeof Mapping)[];
+
+    private mappingNotifiers: Notifier[] = [];
+
+    public constructor() {
+        super();
+        this.columnInternals.sortOperation = TableColumnSortOperation.basic;
+    }
```

Comment:
sortOperation can be passed via getColumnInternalsOptions instead of overriding it here